src/app/firebase.service.ts

```diff
@@ -66,6 +66,23 @@ export class FirebaseService {
       .valueChanges();
   }
 
+  /**
+   * Adds round to firestore and sets round to the session's current round
+   * @param sessionID Session ID to which round will be added
+   * @param roundData data for the round in FirebaseRound interface format
+   */
+  public createRoundInSession(sessionId: string, roundData: FirebaseRound): Promise<RoundPath> {
+    return this.firestore.collection('sessions/' + sessionId + '/rounds').add(roundData).then(doc =>
+      ({sessionId, roundId: doc.id})
+    );
+  }
+
+  public setCurrentRound(roundPath: RoundPath): Promise<RoundPath>{
+    return this.firestore.collection('sessions').doc(roundPath.sessionId).update({currentRoundId: roundPath.roundId}).then(doc =>
+      (roundPath)
```

Comment:
I don't know how useful it is to just return the same `roundPath` that was input to the function since you would already know it.

Reply:
What would you prefer we return instead?

Reply:
What would you suggest instead?

Reply:
Could we do a `Promise<void>`?

Something like

```ts
  public async setCurrentRound(roundPath: RoundPath): Promise<void> {
    await this.firestore.collection('sessions').doc(roundPath.sessionId)
      .update({currentRoundId: roundPath.roundId});
  }
```

(equivalently,)

```ts
  public setCurrentRound(roundPath: RoundPath): Promise<void> {
    return this.firestore.collection('sessions').doc(roundPath.sessionId)
      .update({currentRoundId: roundPath.roundId}).then(() => {});
  }
```

Reply:
You could just return whatever the response is from Firebase, you are always free to ignore it with a `() => {}` kind of thing.
```ts
public setCurrentRound(roundPath: RoundPath) {
    return this.firestore.collection('sessions').doc(roundPath.sessionId)
      .update({currentRoundId: roundPath.roundId});
  }
```

Reply:
Oh, yeah, nifty! :relaxed: